### PR TITLE
Parallax in Darkness

### PR DESCRIPTION
**Pull request by HMBGERDO**
## What Does This PR Do

This PR allows you to toggle parallax in darkness which will render parallax instead of dark tiles if you wear nightvisions or other item that adjusts your vision

## Why It's Good For The Game

This was a bug from widescreen PR when you could seen parallax instead of dark tiles with nightvisions
Bug was fixed because its disorientating but i find this thing really cool and beauty
So here i add this thing back.

## Images of changes

https://youtu.be/JWh8ru8avFc
image

## Testing

compiled, changed settings, saw parallax, no SQL errors

## Changelog
:cl:
add: Added toggleable parallax in darkness.
/:cl:

### CODE REVIEW

**Comment by GDNgit** [comment]
this is certainly a cool idea, but I don't personally love the super bright version we got here, previously the Parallax would be darkened when viewing it from indoors, maybe you could allow for a brightness setting?

**Comment by HMBGERDO** [comment]
> this is certainly a cool idea, but I don't personally love the super bright version we got here, previously the Parallax would be darkened when viewing it from indoors, maybe you could allow for a brightness setting?

It is still darker than space turf parallax and its brightness depends on your nightvision, with thermals parallax would be darker.
Sadly i dont think there is easy way to make darkness on it but i will think about it later

**Review by AffectedArc07 — DISMISSED** [review]
This needs to be a bitflag toggle in the regular `toggles_2` column, not an entire extra DB column for a binary toggle.

**Comment by HMBGERDO** [comment]
No longer requires SQL update, moved to toggles2

File: code/__DEFINES/preferences_defines.dm [hunk 1]
```diff
@@ -62,6 +62,7 @@
 #define PREFTOGGLE_2_MC_TABS				(1<<15) // 32768
 #define PREFTOGGLE_2_DANCE_DISCO			(1<<16) // 65536
 #define PREFTOGGLE_2_MOD_ACTIVATION_METHOD	(1<<17) // 131072
+#define PREFTOGGLE_2_PARALLAX_IN_DARKNESS	(1<<18) // 262144
 
 #define TOGGLES_2_TOTAL 			262144 // If you add or remove a preference toggle above, make sure you update this define with the total value of the toggles combined.
```
Comment (hal9000PR) [inline code/__DEFINES/preferences_defines.dm:67]: change the total. as per the comment

**Review by hal9000PR — CHANGES_REQUESTED** [review]

File: code/modules/client/preference/link_processing.dm [hunk 2]
```diff
@@ -1086,6 +1086,10 @@
 					if(parent && parent.mob && parent.mob.hud_used)
 						parent.mob.hud_used.update_parallax_pref()
 
+				if("parallax_darkness")
+					toggles2 ^= PREFTOGGLE_2_PARALLAX_IN_DARKNESS
+					parent?.mob?.hud_used?.update_parallax_pref()
```
Comment (Contrabang) [inline code/modules/client/preference/link_processing.dm:1091]: If a holder somehow has no client, it should probably runtime while running this.
```suggestion
					parent.mob?.hud_used?.update_parallax_pref()
```

File: code/__DEFINES/preferences_defines.dm [hunk 3]
```diff
@@ -62,8 +62,9 @@
 #define PREFTOGGLE_2_MC_TABS				(1<<15) // 32768
 #define PREFTOGGLE_2_DANCE_DISCO			(1<<16) // 65536
 #define PREFTOGGLE_2_MOD_ACTIVATION_METHOD	(1<<17) // 131072
+#define PREFTOGGLE_2_PARALLAX_IN_DARKNESS	(1<<18) // 262144
 
-#define TOGGLES_2_TOTAL 			262144 // If you add or remove a preference toggle above, make sure you update this define with the total value of the toggles combined.
+#define TOGGLES_2_TOTAL 			524287 // If you add or remove a preference toggle above, make sure you update this define with the total value of the toggles combined.
```
Comment (Contrabang) [inline code/__DEFINES/preferences_defines.dm:67]: Thank you for properly fixing this. For any other reviewers, this number is correct. The formula is `(1 << LATEST+1 ) - 1`

**Review by Contrabang — COMMENTED** [review]
I'm no master of plane masters, but this looks good to me. Just one requested change

**Review by S34NW — APPROVED** [review]

**Pull request merged**